system-addon/content-src/components/TopSites/_TopSites.scss

```diff
@@ -326,13 +321,54 @@
 }
 
 .topsite-form {
+  .form-input-container {
+    display: flex;
```

Comment:
What needs to flex here? I believe oddness happens on narrow when things try to overflow with various other widths specified elsewhere.

Reply:
The `display: flex` declaration moved to the container I mentioned [in this comment](https://github.com/mozilla/activity-stream/pull/3965#discussion_r168810824) that wraps input and preview together. I used it to place both elements as siblings while allowing the preview to have a margin top to align with the input and not shift its sibling.
Do you still think this need to change considering the latest way it's implemented?